infrastructure/messaging/infrastructure/metrics/metrics_collector.py

```
@@ -0,0 +1,242 @@
+import time
+import threading
+from typing import Dict, List, Optional, Tuple
+from collections import deque
+from dataclasses import dataclass, field
```

Flake8 (7.3.0)

[error] 'dataclasses.field' imported but unused

(F401)